compiler/test/compilable/test24504.c

```diff
@@ -0,0 +1,7 @@
+// REQUIRED_ARGS: -os=windows -g
```

Comment:
could you add `-c` here as well, the macOS CI failures seem to resulting from a missing `main` function. I know this is a compilable test and that shouldn't matter, but it would be good to rule out.

Reply:
Yeah okay—that's it pushed now

Reply:
Looks like the same issue with `-c`:
```
2024-04-19T05:02:32.0668560Z  ... compilable/test24504.c         -c -os=windows -g ()
2024-04-19T05:02:32.0673260Z ==============================
2024-04-19T05:02:32.0722390Z Test 'compilable/test24504.c' failed. The logged output:
2024-04-19T05:02:32.0729000Z /Users/runner/work/dmd/dmd/generated/osx/release/64/dmd -conf= -m64 -Icompilable -c -os=windows -g  -od/Users/runner/work/dmd/dmd/compiler/test/test_results/compilable -of/Users/runner/work/dmd/dmd/compiler/test/test_results/compilable/test24504_0.o  -c compilable/test24504.c 
2024-04-19T05:02:32.0772720Z ld: Undefined symbols:
2024-04-19T05:02:32.0843160Z   _main, referenced from:
2024-04-19T05:02:32.0887700Z       <initial-undefines>
2024-04-19T05:02:32.0890780Z clang: error: linker command failed with exit code 1 (use -v to see invocation)
2024-04-19T05:02:32.0899230Z Error: C preprocess command clang failed for file compilable/test24504.c, exit status 1
```

Would a `DISABLED: osx` be acceptable?

Reply:
Yes

Reply:
> Yes

It's now disabled on macOS. It works on FreeBSD so it should work on macOS, barring the preprocessor somehow getting the linker involved.

Reply:
Sounds more like there's a problem with the Mac CI not handling the "compilable" directory properly.

There are a lot of files in compilable that do not have main()

Reply:
Mhm, the same issue happened with the test for [https://github.com/dlang/dmd/pull/16399#discussion_r1572955894](https://github.com/dlang/dmd/pull/16399#discussion_r1572955894)—it seems related to the presence of `-os=windows` on the command-line